core/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -77,6 +90,25 @@ public IndexNameExpressionResolver() {
         return concreteIndices(context, indexExpressions);
     }
 
+    /**
+     * Translates the provided index expression into actual concrete indices.
+     *
+     * @param state             the cluster state containing all the data to resolve to expressions to concrete indices
+     * @param options           defines how the aliases or indices need to be resolved to concrete indices
+     * @param startTime         The start of the request where concrete indices is being invoked for
+     * @param indexExpressions  expressions that can be resolved to alias or index names.
+     * @return the resolved concrete indices based on the cluster state, indices options and index expressions
+     * @throws IndexMissingException if one of the index expressions is pointing to a missing index or alias and the
+     * provided indices options in the context don't allow such a case, or if the final result of the indices resolution
+     * contains no indices and the indices options in the context don't allow such a case.
+     * @throws IllegalArgumentException if one of the aliases resolve to multiple indices and the provided
+     * indices options in the context don't allow such a case.
+     */
+    public String[] concreteIndices(ClusterState state, IndicesOptions options, long startTime, String... indexExpressions) throws IndexMissingException, IllegalArgumentException {
```

Comment:
where is this method called? Looking into why we have so many different concreteIndices methods here :)